modules/auxiliary/scanner/smb/smb_enumshares.rb

```diff
@@ -217,84 +164,103 @@ def get_files_info(ip, _rport, shares, info)
 
     logdata = ''
 
-    list = shares.collect { |e| e[0] }
-    list.each do |x|
-      x = x.strip
-      if (x == 'ADMIN$') || (x == 'IPC$')
+    shares.each do |share|
+      share_name = share[:name].strip
+      if (share_name == 'ADMIN$') || (share_name == 'IPC$')
+        next
+      end
+
+      if (share_name == 'Users') && !datastore['SpiderProfiles']
         next
       end
 
       if !datastore['ShowFiles']
-        print_status("Spidering #{x}.")
+        print_status("Spidering #{share_name}.")
       end
+
+      begin
+        tree = simple.client.tree_connect("\\\\#{ip}\\#{share_name}")
+      rescue RubySMB::Error::UnexpectedStatusCode => e
+        print_error "Error when trying to connect to share #{share_name} - #{e.status_code.name}"
+        print_status "Spider #{share_name} complete."
+        next
+      end
+
       subdirs = ['']
-      if (x.strip == 'C$') && datastore['SpiderProfiles']
-        subdirs = profile_options(ip, x)
+      if (share_name == 'C$') && datastore['SpiderProfiles']
+        subdirs = profile_options(tree, share[:type])
       end
       until subdirs.empty?
-        depth = subdirs[0].count('\\')
-        if datastore['SpiderProfiles'] && (x == 'C$')
-          if depth - 2 > datastore['MaxDepth']
+        depth = subdirs.first.count('\\')
+
+        if share_name == 'C$'
+          if datastore['SpiderProfiles']
+            if (depth - 2) > datastore['MaxDepth']
+              subdirs.shift
+              next
+            end
+          else
             subdirs.shift
             next
           end
-        elsif depth > datastore['MaxDepth']
+        end
+
+        if depth > datastore['MaxDepth']
           subdirs.shift
           next
         end
-        read, write, type, files = eval_host(ip, x, subdirs[0])
+
+        read, write, _type, files = eval_tree(tree, share[:type], subdirs.first)
+
         if files && (read || write)
-          if files.length < 3
+          if files.empty?
             subdirs.shift
             next
           end
-          header = ''
-          if simple.client.default_domain && simple.client.default_name
-            header << " \\\\#{simple.client.default_domain}"
-          end
-          header << "\\#{x.sub('C$', 'C$\\')}" if simple.client.default_name
-          header << subdirs[0]
 
+          header = ''
           pretty_tbl = Rex::Text::Table.new(
             'Header' => header,
             'Indent' => 1,
             'Columns' => [ 'Type', 'Name', 'Created', 'Accessed', 'Written', 'Changed', 'Size' ]
           )
 
-          f_types = {
-            1 => 'RO', 2 => 'HIDDEN', 4 => 'SYS', 8 => 'VOL',
-            16 => 'DIR', 32 => 'ARC', 64 => 'DEV', 128 => 'FILE'
-          }
+          if simple.client.default_domain && simple.client.default_name
```

Comment:
Just confirming, does `default_name` work for ruby_smb 👀

Reply:
In my testing, both `default_name` and `default_domain` us back the server's PC name, e.g. `DESKTOP-T5PBUMF` 👍